services/motion/builtin/builtin.go

```diff
@@ -231,24 +236,111 @@ func (ms *builtIn) MoveOnGlobe(
 	kinematicsOptions.GoalRadiusMM = math.Min(motionCfg.PlanDeviationM*1000, 3000)
 	kinematicsOptions.HeadingThresholdDegrees = 8
 
-	plan, kb, err := ms.planMoveOnGlobe(
-		ctx,
-		componentName,
-		destination,
-		movementSensorName,
-		obstacles,
-		kinematicsOptions,
-		extra,
-	)
-	if err != nil {
-		return false, fmt.Errorf("error making plan for MoveOnMap: %w", err)
-	}
+	movementSensor, ok := ms.movementSensors[movementSensorName]
+	if !ok {
+		return false, resource.DependencyNotFoundError(movementSensorName)
+	}
+
+	// TODO: if we disentangle creation of the kinematic base from the plan call we make we don't need this and can do this once
+	var kinematicBase kinematicbase.KinematicBase = nil
+
+	// Shared variables for planning and execution threads, are all guarded with a mutex
+	var planMu sync.Mutex
+	var moveCtx context.Context
+	var currentPlan [][]referenceframe.Input = nil
+
+	// Shared variable between threads that can trigger replanning and planning thread
+	var replanFlag atomic.Bool
+	replanFlag.Store(true)
+
+	var successFlag atomic.Bool
+	successFlag.Store(false)
+	errChan := make(chan error)
+
+	ms.backgroundWorkers.Add(1)
+	goutils.ManagedGo(func() {
+
+		// maybe move this into the restart tickers function
+		cancelCtx, cancelFn := context.WithCancel(context.Background())
+		ms.cancelFn = cancelFn
+
+		positionPollingTicker := time.NewTicker(time.Duration(1000/motionCfg.PositionPollingFreqHz) * time.Millisecond)
+		obstaclePollingTicker := time.NewTicker(time.Duration(1000/motionCfg.ObstaclePollingFreqHz) * time.Millisecond)
+
+		for !successFlag.Load() && ctx.Err() == nil {
+			select {
+			case <-positionPollingTicker.C:
+				// TODO: theres nothing stopping us from making another thread before the previous has completed
+				ms.backgroundWorkers.Add(1)
+				goutils.ManagedGo(func() {
+					fmt.Println("position poll")
+					current, _, err := movementSensor.Position(cancelCtx, nil)
+					if err != nil {
+						errChan <- err
+						return
+					}
+					if spatialmath.GeoPointToPose(current, destination).Point().Norm() <= kinematicsOptions.PlanDeviationThresholdMM {
```

Comment:
Instead of checking this inside of the position polling thread, what I would propose is:
-execute the planned motion plan in its entirety
-once the plan has completed and motion has ceased, *then* check if our location is within deviation of the goal
-if not, execute a replan.

The above would only replan if a plan execution falls short of the goal, as an overshoot or side-skew while still moving would trigger the usual replanning.

Reply:
That makes sense to me.

Reply:
Yeah I went back and forth a few times on how to do the exit condition.  I like this proposal

Reply:
Implemented this, going to resolve this thread